src/django_components/app_settings.py

```diff
@@ -93,95 +114,134 @@ class ContextBehavior(str, Enum):
+class ComponentsSettings(NamedTuple):
+    autodiscover: Optional[bool] = None
+    dirs: Optional[Sequence[Union[str, PathLike, Tuple[str, str], Tuple[str, PathLike]]]] = None
+    app_dirs: Optional[Sequence[str]] = None
+    context_behavior: Optional[ContextBehaviorType] = None
+    dynamic_component_name: Optional[str] = None
+    libraries: Optional[List[str]] = None
+    multiline_tags: Optional[bool] = None
+    reload_on_template_change: Optional[bool] = None
+    static_files_allowed: Optional[List[Union[str, re.Pattern]]] = None
+    forbidden_static_files: Optional[List[Union[str, re.Pattern]]] = None
+    tag_formatter: Optional[Union["TagFormatterABC", str]] = None
+    template_cache_size: Optional[int] = None
+
+
+# NOTE: Some defaults depend on the Django settings, which may not yet be
+# initialized at the time that these settings are generated. For such cases
+# we define the defaults as a factory function, and use the `Dynamic` class to
+# mark such fields.
+@dataclass(frozen=True)
+class Dynamic(Generic[T]):
+    getter: Callable[[], T]
+
+
+# This is the source of truth for the settings defaults. If the documentation
+# does NOT match it, the documentation should be updated.
+#
+# NOTE: Because we need to access Django settings to generate default dirs
+#       for `COMPONENTS.dirs`, we do it lazily.
+# NOTE 2: We show the defaults in the documentation, together with the comments
+#        (except for the `Dynamic` instances and comments like `type: ignore`).
+#        So `fmt: off` turns off Black formatting and `snippet:defaults` allows
+#        us to extract the snippet from the file.
+#
+# fmt: off
+# --snippet:defaults--
+defaults = ComponentsSettings(
+    autodiscover=True,
+    context_behavior=ContextBehavior.DJANGO.value,  # "django" | "isolated"
+    # Root-level "components" dirs, e.g. `/path/to/proj/components/`
+    dirs=Dynamic(lambda: [Path(settings.BASE_DIR) / "components"]),  # type: ignore[arg-type]
+    # App-level "components" dirs, e.g. `[app]/components/`
+    app_dirs=["components"],
+    dynamic_component_name="dynamic",
+    libraries=[],  # E.g. ["mysite.components.forms", ...]
+    multiline_tags=True,
+    reload_on_template_change=False,
+    static_files_allowed=[
+        ".css",
+        ".js", ".jsx", ".ts", ".tsx",
+        # Images
+        ".apng", ".png", ".avif", ".gif", ".jpg",
+        ".jpeg", ".jfif", ".pjpeg", ".pjp", ".svg",
+        ".webp", ".bmp", ".ico", ".cur", ".tif", ".tiff",
+        # Fonts
+        ".eot", ".ttf", ".woff", ".otf", ".svg",
+    ],
+    forbidden_static_files=[
+        # See https://marketplace.visualstudio.com/items?itemName=junstyle.vscode-django-support
+        ".html", ".django", ".dj", ".tpl",
+        # Python files
+        ".py", ".pyc",
+    ],
+    tag_formatter="django_components.component_formatter",
+    template_cache_size=128,
+)
+# --endsnippet:defaults--
+# fmt: on
+
+
+class InternalSettings:
```

Comment:
I also renamed `AppSettings` to `InternalSettings`, so there's no confusion beteen this and `ComponentsSettings`